qiita_pet/handlers/study_handlers/prep_template.py

```diff
@@ -77,3 +80,43 @@ def post(self):
         else:
             raise HTTPError(400, 'Unknown prep template action: %s' % action)
         self.write(result)
+
+
+class PrepFilesHandler(BaseHandler):
+    @authenticated
+    def get(self):
+        study_id = self.get_argument('study_id')
+        prep_file = self.get_argument('prep_file')
+        prep_type = self.get_argument('type')
+
+        # TODO: Get file types for the artifact type
+        # FILE TYPE IN POSTION 0 MUST BE DEFAULT FOR SELECTED
+        file_types = supported_filepath_types(prep_type)
+
+        selected = []
+        not_selected = []
+        _, base = get_mountpoint("uploads")[0]
+        uploaded = get_files_from_uploads_folders(study_id)
+        prep = pd.read_table(join(base, study_id, prep_file), sep='\t')
+        if 'run_prefix' in prep.columns:
+            # Use run_prefix column of prep template to auto-select
+            # per-sample uploaded files if available.
+            per_sample = True
+            prep_prefixes = set(prep['run_prefix'])
+            for _, filename in uploaded:
+                for prefix in prep_prefixes:
+                    if filename.startswith(prefix):
+                        selected.append(filename)
+                    else:
+                        not_selected.append(filename)
+        else:
+            per_sample = False
+            not_selected = [f for _, f in uploaded]
+
+        # Write out if this prep template supports per-sample files, and the
+        # as well as pre-selected and remaining files
+        self.write({
+            'per_sample': per_sample,
```

Comment:
regarding the name of the variable, and the name of the keyword in the output dictionary.